### PR TITLE
Brotli Update Readme

### PR DESCRIPTION
**Pull request by Vedin**
@ahsonkhan we can continue conversation here. I think that I resolved all comments from previous pr.

### CODE REVIEW

**Comment by Vedin** [comment]
@dotnet-bot test Innerloop Ubuntu16.04 Debug Build and Test

File: src/System.IO.Compression.Brotli/README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ to deflate but offers more dense compression.
 
 The specification of the Brotli Compressed Data Format is defined in [RFC 7932](https://www.ietf.org/rfc/rfc7932.txt).
 
-Brotli encoding is supported by most web browsers, major web servers, and some CDNs.
+Brotli encoding is supported by most web browsers, major web servers, and some CDNs (Content delivery network).
```
Comment (ahsonkhan) [inline src/System.IO.Compression.Brotli/README.md]: nit: Content Delivery Networks?

File: src/System.IO.Compression.Brotli/README.md [hunk 2]
```diff
@@ -29,43 +29,41 @@ public static class Brotli {
...
-The ```Compress``` performs data compression. Taking data from ```source``` and writing compressed data to ```destination```.
+The ```Compress``` performs data compression. It takes data from ```source``` and writes compressed data to ```destination```.
 
 The ```FlushEncoder``` returns compressed data, which was sent to ```state``` using Compress. This method should always be called after all Compress executions (```isFinished = true```) or when you want to get compressed data immediately (```isFinished = false```).
 
-The ```Decompress``` Decompresses the data from ```source``` into ```destination```. 
+The ```Decompress``` decompresses the data from ```source``` into ```destination```. 
 
-```SetQuality``` allows you to set quality of compression ```0 to 11```. The higher quality means the higher compression ratio, but more compute time.
+```SetQuality``` allows you to set quality of compression ```between 0 to 11```. The higher quality means the higher compression ratio, but more compute time.
```
Comment (ahsonkhan) [inline src/System.IO.Compression.Brotli/README.md]: "allows you to set the quality of compression between 0 and 11"
I would also add "inclusive" to be less ambiguous. Between 0 to 11 could mean [0, 11] or (0, 11).

File: src/System.IO.Compression.Brotli/README.md [hunk 3]
```diff
@@ -29,43 +29,41 @@ public static class Brotli {
...
 
-```SetWindow``` - Logarithm of recommended sliding LZ77 window size. Encoder may reduce this value, e.g. if input is much smaller than window size. Window size is (1 << value) - 16. Possible values: ```11 to 24```
+```SetWindow``` - Logarithm of recommended sliding LZ77 window size. Encoder may reduce this value, e.g. if input is much smaller than window size. Window size is ```(1 << value) - 16```. Possible values: ```11 to 24```
```
Comment (ahsonkhan) [inline src/System.IO.Compression.Brotli/README.md]: I am curious, is there any difference between having three \` and just a single \`?

i.e. 
\```(1 << value) - 16\```
vs.
\`(1 << value) - 16\`

Doesn't seem like it:
```(1 << value) - 16```
`(1 << value) - 16`

File: src/System.IO.Compression.Brotli/README.md [hunk 4]
```diff
@@ -29,43 +29,41 @@ public static class Brotli {
...
 Simple method to compress bytes to file.
 ```C#
 
- static void CompressSimple(byte[] bytes, string OutFile)
+ static void CompressSimple(byte[] bytes, string outFile)
  {
     Brotli.State state = new Brotli.State();
+    Span<byte> spanBytes = new Span<byte>(bytes);
     byte[] compressed = new byte[Brotli.GetMaximumCompressedSize(bytes.Length)];
-    state.SetQuality();
```
Comment (ahsonkhan) [inline src/System.IO.Compression.Brotli/README.md]: Was the SetQuality call unnecessary?

File: src/System.IO.Compression.Brotli/README.md [hunk 5]
```diff
@@ -29,43 +29,41 @@ public static class Brotli {
...
-    TransformationStatus result = Brotli.Compress(bytes, compressed, out int consumed, out int written, ref state);
+    TransformationStatus result = Brotli.Compress(spanBytes, compressed, out int consumed, out int written, ref state);
     while (result != TransformationStatus.Done)
     {
-        result = Brotli.Compress(bytes, compressed, out consumed, out written, ref state);
-        bytes = bytes.AsSpan().Slice(consumed).ToArray();
+        result = Brotli.Compress(spanBytes, compressed, out consumed, out written, ref state);
+        spanBytes = spanBytes.Slice(consumed);
```
Comment (ahsonkhan) [inline src/System.IO.Compression.Brotli/README.md]: Is this assignment necessary?
Why not pass `spanBytes.Slice(consumed)` to Brotli.Compress?

File: src/System.IO.Compression.Brotli/README.md (hunk 4 above)
Comment (Vedin) [inline src/System.IO.Compression.Brotli/README.md]: Yes, setXXX is no sense method, so I removed it.

**Comment by Vedin** [comment]
@dotnet-bot test Innerloop Ubuntu16.04 Release Build and Test

File: src/System.IO.Compression.Brotli/README.md (hunk 5 above)
Comment (Vedin) [inline src/System.IO.Compression.Brotli/README.md]: as I remember Slice return Span and don't change argument. So if we have multiply iterations of this loop we had or create a variable and do ```allConsumed += consumed``` and than can do ```spanBytes.Slice(allConsumed)``` or just update spanBytes every time.

Comment (ahsonkhan) [inline src/System.IO.Compression.Brotli/README.md]: That is correct. Good point.

File: src/System.IO.Compression.Brotli/README.md (hunk 3 above)
Comment (Vedin) [inline src/System.IO.Compression.Brotli/README.md]: Yes, really. Should I change this everywhere?

Comment (ahsonkhan) [inline src/System.IO.Compression.Brotli/README.md]: It seems unnecessary to have three \`. I would change it to single \` but your call.

File: src/System.IO.Compression.Brotli/README.md (hunk 2 above)
Comment (ahsonkhan) [inline src/System.IO.Compression.Brotli/README.md]: to clarify: "allows you to set **the** quality of compression between 0 **and** 11"

**Comment by ahsonkhan** [comment]
@Vedin, thoughts on adding a reference link?

https://github.com/dotnet/corefxlab/pull/1658#discussion_r126804291

**Comment by Vedin** [comment]
@ahsonkhan , Do you mean a reference to a google brotli repo?
In fact, I don't know. Can we do it or should we do it?

**Comment by ahsonkhan** [comment]
@Vedin, @shiftylogic good to go?

**Pull request merged**